### PR TITLE
Tidy up the heat equation example

### PR DESCRIPTION
**Pull request by sbastrakov**
Add correction check as return code, as opposed to just printing earlier. Minor style fixes. Update the header which was missed in the original PR #978 .

### CODE REVIEW

File: example/heatEquation/src/heatEquation.cpp [hunk 1]
```diff
@@ -1,4 +1,5 @@
-/* Copyright 2019 Benjamin Worpitz, Matthias Werner
+/* Copyright 2020 Benjamin Worpitz, Matthias Werner, Jakob Krude,
+ *                Sergei Bastrakov
```
Comment (sbastrakov) [inline example/heatEquation/src/heatEquation.cpp:2]: Note: this change of the header was just forgotten in the original PR. This example was developed by @jkrude and me, based on the existing ones.

**Comment by sbastrakov** [comment]
@BenjaminW3 in my last commit 2017bfe6d32cdaab7d86bea466d5dc9bee34f87f I tried to set up filters, to investigate only the 3 CI combinations originally failing. But it seems all combinations were ran again. Could you take a look if I did something wrong?

**Comment by BenjaminW3** [comment]
Maybe #1078 broke this feature. I will have to investigate...

**Review by BenjaminW3 — APPROVED** [review]

**Comment by BenjaminW3** [comment]
@sbastrakov I took a look at the `ci_filter` feature again and it seems to work.
The confusing part is that skipped jobs are still shown and they are shown as successful.
I can not make them disappear, but I can make them show up as failed.

**Comment by sbastrakov** [comment]
@BenjaminW3 thanks for looking and explaining. I think in this case it is less confusing to mark skipped as successful than failed (like done now).

**Comment by BenjaminW3** [comment]
The main reason for making them fail now is because else this would allow to make arbitrary PRs green and mergable even when they still have issues. It is not easy for reviewers to see if there are skipped jobs.

**Pull request merged**